src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/Models/DependencyModel.cs

```diff
@@ -42,6 +42,12 @@ internal class DependencyModel : IDependencyModel
             {
                 Flags = Flags.Except(DependencyTreeFlags.SupportsRemove);
             }
+
+            if (Properties.TryGetValue("Visible", out string visibleMetadata)
```

Comment:
Both hierarchy and graph nodes utilize this path, and I've validated that the flag is respected on `Reference` and `ProjectReference` items.

Graph node items are harder to validate--I'm still figuring out how to do this. Of course, it's difficult to even add the `Visible` metadata to the items that back graph nodes.